### PR TITLE
fix: remove async for createProgramAddress and findProgramAddress (#2…

### LINKED ISSUE #23184: findProgramAddress does not need to be async
#### Problem
findProgramAddress is an async function but nothing about it requires it to be async
This makes it harder to write programs, for example if you wanted to use it in an array map you would have to use promise.all

For example if I wanted to find all the PDA for an account I would need to do something like the below
```js
const findAllDerivedAccounts = await Promise.all(accounts.map(async (account, index) => {
  const [PDA] = await findProgramAddress(account.toBuffer(), programId);

  return PDA; 
}))
```

#### Proposed Solution
Remove async from **createProgramAddress** and **findProgramAddress**

### PR DESCRIPTION
**Pull request by yangli-io**
Fixes #23184

### CODE REVIEW

**Comment by t-nelson** [comment]
Was the sha256 library changed to be sync at some point?  That used to be what forced this family of methods to be async.

You can't break the API like this.  Please add a sync variant instead.

**Comment by yangli-io** [comment]
Thanks for the feedback, I forgot it will break people using promises. Updated to allow for a sync version

**Comment by Arrowana** [comment]
How breaking is this though? Code can await a non `async` method and awaiting with `.then` is still possible
Or is it just a matter of not touching the API types even if non technically "breaking"?

**Comment by yangli-io** [comment]
it would be breaking if the developer has wrote their code as such

```
PublicKey.findProgramAddress([Buffer.from('', 'utf8')], programId).then();
```

I dont like the async versions but yes it would cause some issues. Probably be good make it completely sync in a major upgrade.

File: web3.js/src/publickey.ts [hunk 1]
```diff
@@ -143,10 +143,10 @@ export class PublicKey extends Struct {
    * Derive a program address from seeds and a program ID.
    */
   /* eslint-disable require-await */
```
Comment (joncinque) [inline web3.js/src/publickey.ts:145]: nit: Probably don't need this comment, right?  Makes me wonder if this needed to be async for some reason

**Review by joncinque — COMMENTED** [review]
This looks good to me pending one tiny nit, and would be a useful addition.  @t-nelson do you have any issues with this?

**Review by t-nelson — APPROVED** [review]
> @t-nelson do you have any issues with this?

Nope!  I'm happy once you're happy

**Pull request merged**